azerothcore/azerothcore-wotlk
# feat(Core/Locale): implement separated manager for game locale from ObjectMgr

## Changes Proposed:
- Move all localizations code to `GameLocale`
- Add a new system for localizing modules
- Add support locale for `autobroadcast`
- Add support locale for `commands.help`
- Implement separated manager for `autobroadcast` - #7420
- Add localization for all game `races` and `classes` (now using in command `.pinfo`)
- Sync naming `npc_text` and fix delay fields naming (https://github.com/TrinityCore/TrinityCore/commit/c574972cb2d000caacd155ddd32af425bed259ee)
- Sync `broadcast_text` naming (https://github.com/TrinityCore/TrinityCore/commit/6c85964065dc2b2d397904d79755eacc43a8ab0d)
- Add support `fmt` style for all functions using db table `acore_string`

## Issues Addressed:
- https://github.com/azerothcore/mod-npc-talent-template/pull/32#issuecomment-827184388

# Tests Performed:
- `autobroadcast_locale`
![image](https://user-images.githubusercontent.com/18329778/138873856-e7c50d80-0f17-4ba6-9456-a50be2330909.png)
![image](https://user-images.githubusercontent.com/18329778/138878529-0be2eb9b-bcac-436a-a4db-bb5a3ee5955e.png)

- `mod-player-info-at-login` with module locale system
![image](https://user-images.githubusercontent.com/18329778/138878683-b1675764-1805-455f-be96-1fa3322a111d.png)

- `commands_help_locale`
![image](https://user-images.githubusercontent.com/18329778/138882435-c16995e5-0cc6-444c-b0fa-c34a4af22efd.png)

- `.pinfo` with localize races and classes
![image](https://user-images.githubusercontent.com/18329778/138884816-dd8fb316-0999-40b9-bc8b-345ce9aefb61.png)

## How to Test the Changes:
- Add new module - https://github.com/WarheadCore/mod-player-info-at-login
- Check info at login
- Try test default localization
![image](https://user-images.githubusercontent.com/18329778/138885467-fc2de03a-72e4-4c61-8483-6a18b02cf368.png)

## Known Issues and TODO List:
- [x] #6893
- [x] #6243
- [x] Add example
- [ ] https://github.com/azerothcore/Eluna/pull/10
- [ ] https://github.com/azerothcore/mod-transmog/pull/57
- [x] StringFormat
- [x] `acore_string` 

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
